pennylane/workflow/get_gradient_fn.py

```diff
@@ -26,7 +27,8 @@
 )
 
 
-# pylint: disable=too-many-return-statements, unsupported-binary-operation
+# pylint: disable=too-many-return-statements, unsupported-binary-operation, inconsistent-return-statements
+@debug_logger
```

Comment:
Potentially out of scope, but we should be able to make `tape` a mandatory argument now, since we aren't needing to worry about backward compatibility anymore.